chall01/gmolin.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/python
+import sys
+
+CODE = {'A': '.-',     'B': '-...',   'C': '-.-.', 
+        'D': '-..',    'E': '.',      'F': '..-.',
+        'G': '--.',    'H': '....',   'I': '..',
+        'J': '.---',   'K': '-.-',    'L': '.-..',
+        'M': '--',     'N': '-.',     'O': '---',
+        'P': '.--.',   'Q': '--.-',   'R': '.-.',
+     	'S': '...',    'T': '-',      'U': '..-',
+        'V': '...-',   'W': '.--',    'X': '-..-',
+        'Y': '-.--',   'Z': '--..',
+        
+        '0': '-----',  '1': '.----',  '2': '..---',
+        '3': '...--',  '4': '....-',  '5': '.....',
+        '6': '-....',  '7': '--...',  '8': '---..',
+        '9': '----.',  ' ': ' ',
+        }
+
+def main():
+    nb = len(sys.argv)
+
+    if nb == 2:
+        msg = sys.argv[1]
+        if msg == "":
+            print "usage: ./gmolin.py <a-zA-Z string>"
+            sys.exit()
+        for char in msg:
+            if char.upper() in CODE:
+                continue
+            else:
+                print "usage: ./gmolin.py <a-zA-Z string>"
+                sys.exit()
+        for char in msg:
+            print CODE[char.upper()],
```

Comment:
Like in C, you should avoid print char by char but print a whole string in once.